src/map.cpp

```diff
@@ -618,6 +618,60 @@ void map::move_vehicle( vehicle &veh, const tripoint &dp, const tileray &facing
             veh.skidding = true;
             veh.turn( coll_turn );
         }
+        // accept new position
+        // if submap changed, we need to process grid from the beginning.
+        // TODO: Allow vehicles to get displaced vertically
+        const tripoint &veh_start=veh.global_pos3();
+        for( const int plow_id : veh.all_parts_with_feature( "PLOW" ) ){
+            if( !veh.plow_on ){
+                break;
+            }
+            //draw a line between the start point and end point of where the plow ends up, and try to till it if it is diggable.
+            const tripoint start_plow(veh.parts[plow_id].precalc[0].x + veh_start.x,
+                                      veh.parts[plow_id].precalc[0].y + veh_start.y,
+                                      veh_start.z);
+            const std::vector<tripoint> & plow_line = line_to(start_plow,start_plow+dp);
+            for( const tripoint& plow_pt : plow_line ){
+                if( has_flag("DIGGABLE", plow_pt) ){
+                    ter_set( plow_pt, t_dirtmound );
+                }else{
+                    const int speed = dp.x + dp.y;
```

Comment:
Yes.
A fast vehicle will have multiple invocations of `vehproceed`, each invoking `move_vehicle`. `dp` will always have x and y either -1, 0 or 1. Combinations like x=1 and y=-1 are possible, but x=2 shouldn't happen.